### ISSUE TITLE
Prediction clipping sets background values to 0.5 instead of 0

### ISSUE THREAD

**Issue by valosekj** [sct-pipeline/contrast-agnostic-softseg-spinalcord#71]
MONAI prediction clipping sets background values to 0.5 instead of 0! 

https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/f88d927c6709dcebb79d38ccd3bd45aa2e355474/monai/run_inference_single_image.py#L308-L309

<img width="300" alt="image" src="https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/assets/39456460/1f25d17a-2122-495d-a188-430264f9bdeb">

---

Possible fix:

```python
# clip the prediction between 0.5 and 1
pred = torch.clamp(pred, 0.5, 1)
# set background values to 0
pred[pred <= 0.5] = 0
```

or `sct_maths` command for already existing predictions:

```console
$ sct_maths -i ${file}_seg_monai.nii.gz -thr 0.51 -o ${file}_seg_monai_thr.nii.gz
```

**Comment by naga-karthik** [comment]
Thanks for catching this bug, Jan! It has been fixed in commit 4694169352fb962f607e7684fe778677adc123dc !

**Comment by valosekj** [comment]
Thank you for the fix, Naga!
